editor/store/selectors.js

```diff
@@ -1538,6 +1538,7 @@ export const getInserterItems = createSelector(
 		state.settings.allowedBlockTypes,
 		state.settings.templateLock,
 		state.sharedBlocks.data,
+		getBlockTypes(),
```

Comment:
This gets me thinking: Elsewhere we depend on selectors who themselves are cached, we often duplicate those dependencies into the second argument set, which is why [with `rememo@3.0.0` I added a `getDependants` function](https://github.com/aduth/rememo#api) to protect against such a piercing of abstraction. But, if we can assume that invoking the selector is at most only marginally less trivial than spreading in its dependants (as we're doing here), maybe that can also be a fine substitute?

It's a more complex problem here because `getBlockTypes` is not the selector itself, and we'd have no way to provide `state` to the selector form of `getBlockTypes.getDependants`.

Which is all to say: This is probably the correct approach here, but can / should we consider adopting this pattern more widely for composing memoized selectors?

Reply:
Hi @aduth,
Here, unfortunately, getDependants is not an option because to the editor store we are just calling an external function.
Regarding the use of getDependants, the decision should be taken on a case by case basis, but in general I feel when the selector is cached we should use getDependants, the cases where we may call the selector directly when computing the dependents are when the selector is trivial and it is not cached so it makes sense to call it.